hrabsence/tests/phpunit/CRM/HRAbsence/BAO/HRAbsenceTypeTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_HRAbsence_Test_Fabricator_HRAbsenceType as AbsenceTypeFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobContract as HRJobContractFabricator;
+use CRM_Hrjobcontract_Test_Fabricator_HRJobLeave as HRJobLeaveFabricator;
+use CRM_HRCore_Test_Fabricator_Activity as ActivityFabricator;
+
+/**
+ * Class CRM_HRAbsence_BAO_HRAbsenceTypeTest
+ * 
+ * @group headless
+ */
+class CRM_HRAbsence_BAO_HRAbsenceTypeTest extends PHPUnit_Framework_TestCase 
+  implements HeadlessInterface, TransactionalInterface {
+  
+  protected $absenceType;
+  protected $contact;
+  protected $jobContract;
+  protected $leave;
+  
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->install('uk.co.compucorp.civicrm.hrcore')
+      ->install('org.civicrm.hrjobcontract')
+      ->installMe(__DIR__)
+      ->apply();
+  }
+  
+  /**
+   * Creates absence type, contact, contract and leave entitlement to be used in
+   * tests.
+   */
+  protected function setUp() {
+    $this->absenceType = AbsenceTypeFabricator::fabricateUsingAPI();
+
+    $this->contact = ContactFabricator::fabricate();
+
+    $this->jobContract = HRJobContractFabricator::fabricate(
+      ['contact_id' => $this->contact['id']], 
+      ['period_start_date' => '2015-01-01']
+    );
+
+    $this->leave = HRJobLeaveFabricator::fabricate([
+      'jobcontract_id' => $this->jobContract['id'],
+      'leave_type' => $this->absenceType['id']
+    ]);
+  }
+  
+  /**
+   * Tests that absence typs used in leave requests cannot be deleted.

```

Comment:
Since now the test name is descriptive enough, no need for the description on the docblock